File: php/features.php
<?php
declare(strict_types=1);

// FreeMusic SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeMusicFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeMusicBaseFeature();
            case "test":
                return new FreeMusicTestFeature();
            default:
                return new FreeMusicBaseFeature();
        }
    }
}

File: py/features.py
# FreeMusic SDK feature factory

from feature.base_feature import FreeMusicBaseFeature
from feature.test_feature import FreeMusicTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeMusicBaseFeature(),
        "test": lambda: FreeMusicTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
